action.yml

```diff
@@ -74,24 +74,38 @@ inputs:
     description: The directory containing compile_commands.json file.
     required: false
     default: ""
-outputs:
-  checks-failed:
-    description: An integer that can be used as a boolean value to indicate if all checks failed.
+  quiet-fail:
+    description: A flag that can be used to exit with a non-successfull error code (not '0') if any checks failed.
+    required: false
+    default: true
 runs:
-  using: "docker"
-  image: "Dockerfile"
-  env:
-    USING_CLANG_TOOLS_DOCKER: 'true'
-  args:
-    - --style=${{ inputs.style }}
-    - --extensions=${{ inputs.extensions }}
-    - --tidy-checks=${{ inputs.tidy-checks }}
-    - --repo-root=${{ inputs.repo-root }}
-    - --version=${{ inputs.version }}
-    - --verbosity=${{ inputs.verbosity }}
-    - --lines-changed-only=${{ inputs.lines-changed-only }}
-    - --files-changed-only=${{ inputs.files-changed-only }}
-    - --thread-comments=${{ inputs.thread-comments }}
-    - --ignore=${{ inputs.ignore }}
-    - --database=${{ inputs.database }}
-    - --file-annotations=${{ inputs.file-annotations }}
+  using: "composite"
+  steps:
+    - name: Install action dependencies
+      # FIXME: when cpp-linter released to pypi, use that.
+      shell: bash
+      run: python3 -m pip install clang-tools cpp-linter
+    - name: Install clang-tools binary executables
+      shell: bash
+      run: clang-tools -i ${{ inputs.version }}
+    - name: Run cpp-linter
+      id: cpp-linter
+      shell: bash
+      run: |
+        cpp-linter \
+          --style="${{ inputs.style }}" \
+          --extensions=${{ inputs.extensions }} \
+          --tidy-checks="${{ inputs.tidy-checks }}" \
+          --repo-root=${{ inputs.repo-root }} \
+          --version=${{ inputs.version }} \
+          --verbosity=${{ inputs.verbosity }} \
+          --lines-changed-only=${{ inputs.lines-changed-only }} \
+          --files-changed-only=${{ inputs.files-changed-only }} \
+          --thread-comments=${{ inputs.thread-comments }} \
+          --ignore="${{ inputs.ignore }}" \
+          --database=${{ inputs.database }} \
+          --file-annotations=${{ inputs.file-annotations }}
+    - name: set output variable
+      if: steps.cpp-linter.outputs.checks-failed != 0 && inputs.quiet-fail != 'true'
```

Comment:
Do we need `steps.cpp-linter.outputs.checks-failed != 0`?, `checks-failed` seems has been removed.

Reply:
The variable is output by the composite step, and we can read it in another composite step. But the user's workflow doesn't seem to get access to the output variable. I tried propagating the output variable, but that didn't work. Actually, I think [there's a bug in the composite runners](https://github.com/actions/runner/issues/1742) that prevents output variables getting passed to the calling workflow.

So, this is currently still needed until the cpp-linter pkg starts using an exit code instead of a output variable.

Reply:
> But the user's workflow doesn't seem to get access to the output variable

In this [example](https://github.com/shenxianpeng/hello-world-composite-action/runs/7782344868?check_suite_focus=true) user's workflow could get access to the output variable.

Reply:
What happens if the composite workflow uses a second (or multiple) log command(s)?

The problem here is consistency. I'm not sure the log commands are working as they should.
Compare these two workflows's logs that use the same commit from this branch:
1. file-annotations are shown in the workflow summary (expected behavior): https://github.com/cpp-linter/test-cpp-linter-action/actions/runs/2937479015
2. file-annotations are not shown in the workflow summary (unexpected behavior): https://github.com/cpp-linter/cpp-linter-action/runs/8046787983?check_suite_focus=true

Reply:
OK, I want to make some revert changes to test and understand how it works

Reply:
You can do a `git reset` on this branch if you want. None of the changes I pushed here are too complex.

Reply:
![image](https://user-images.githubusercontent.com/3353385/187010923-ca35535c-cad4-4da3-9a8b-6ee5451311ff.png)

![image](https://user-images.githubusercontent.com/3353385/187010941-99a38f9f-a4d1-498b-8703-062693570a8d.png)

The `latest` tag does not point to the latest commit hash id, it looks like another problem.

Reply:
`af9cb80` was a merge commit of `b4ee98a` + `92298bd`(master branch lastest commit), it seems OK.

![image](https://user-images.githubusercontent.com/3353385/187016327-f5c9c9d4-e203-42ab-acbd-2808adfaddee.png)